registry/handlers/manifests.go

```diff
@@ -142,6 +143,8 @@ func (imh *manifestHandler) GetManifest(w http.ResponseWriter, r *http.Request)
 	if err != nil {
 		if _, ok := err.(distribution.ErrManifestUnknownRevision); ok {
 			imh.Errors = append(imh.Errors, errcode.ErrorCodeManifestUnknown.WithDetail(err))
+		} else if checkAllErrorsStatusCode(err, 401) {
```

Comment:
This doesn't sit well with me. We've introduced a numeric literal (instead of, say, `http. StatusUnauthorized`), but in general, we've "leaked" HTTP code into manifest handler code. We should make this code return registry error that's handled in the app handler.